### PR TITLE
Add documentation for ASN1_INTEGER_new() and ASN1_INTEGER_free()

### PR DESCRIPTION
**Pull request by NMorozxov**
##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/ASN1_INTEGER_new.pod [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ #include <openssl/asn1.h>
+
+ extern const ASN1_INTEGER_it;
+ ASN1_INTEGER * ASN1_INTEGER_new(void);
+ void ASN1_INTEGER_free(ASN1_INTEGER *a);
+
+=head1 DESCRIPTION
+
+ASN1_INTEGER_new() returns an allocated B<ASN1_INTEGER> structure. Its type
+is undefined.
```
Comment (richsalz) [inline doc/man3/ASN1_INTEGER_new.pod:19]: opaque not undefined.

Comment (beldmit) [inline doc/man3/ASN1_INTEGER_new.pod:19]: Sorry, I don't understand what does it mean.

Comment (NMorozxov) [reply doc/man3/ASN1_INTEGER_new.pod:19]: it's a copy/pas issue

Comment (NMorozxov) [reply doc/man3/ASN1_INTEGER_new.pod:19]: fixed

**Comment by NMorozxov** [comment]
@richsalz @beldmit fix commited

**Comment by slontis** [comment]
Please run the following when doing any document updates.
```
>make doc-nits 
```

File: doc/man3/ASN1_INTEGER_new.pod [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+ASN1_INTEGER_new, ASN1_INTEGER_free, ASN1_INTEGER_it -
+ASN1_INTEGER allocation functions
+
+=head1 SYNOPSIS
+
+ #include <openssl/asn1.h>
+
+ extern const ASN1_INTEGER_it;
+ ASN1_INTEGER * ASN1_INTEGER_new(void);
```
Comment (slontis) [inline doc/man3/ASN1_INTEGER_new.pod:13]: *ASN1_INTEGER_new(void);

Comment (NMorozxov) [reply doc/man3/ASN1_INTEGER_new.pod]: fixed

**Comment by NMorozxov** [comment]
@slontis fix commited

File: doc/man3/ASN1_INTEGER_new.pod [hunk 3]
```diff
@@ -0,0 +1,44 @@
+=pod
+
+=head1 NAME
+
+ASN1_INTEGER_new, ASN1_INTEGER_free, ASN1_INTEGER_it -
```
Comment (mattcaswell) [inline doc/man3/ASN1_INTEGER_new.pod:5]: In spite of listing `ASN1_INTEGER_it` here it  doesn't say anything about this type on this page.

File: doc/man3/ASN1_INTEGER_new.pod [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+ #include <openssl/asn1.h>
+
+ extern const ASN1_INTEGER_it;
+ ASN1_INTEGER *ASN1_INTEGER_new(void);
+ void ASN1_INTEGER_free(ASN1_INTEGER *a);
+
+=head1 DESCRIPTION
+
+ASN1_INTEGER_new() returns an allocated B<ASN1_INTEGER> structure.
```
Comment (mattcaswell) [inline doc/man3/ASN1_INTEGER_new.pod:20]: But what *is* an ASN1_INTEGER structure? It should say something about what it represents.

File: doc/man3/ASN1_INTEGER_new.pod [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+=head1 SYNOPSIS
+
+=for openssl generic
+
+ #include <openssl/asn1.h>
+
+ extern const ASN1_INTEGER_it;
+ ASN1_INTEGER *ASN1_INTEGER_new(void);
+ void ASN1_INTEGER_free(ASN1_INTEGER *a);
```
Comment (mattcaswell) [inline doc/man3/ASN1_INTEGER_new.pod:16]: I do wonder whether we would be better having a single "TYPE" page that defines these functions for all the similar ASN1 functions, i.e. something similar to the d2i_X509.pod page, but for new/free etc.

https://github.com/openssl/openssl/blob/master/doc/man3/d2i_X509.pod

Comment (beldmit) [reply doc/man3/ASN1_INTEGER_new.pod:16]: Well, much better!

Comment (NMorozxov) [reply doc/man3/ASN1_INTEGER_new.pod:16]: i must drop this page and combit it with others ? Or I must fix this PR

Comment (beldmit) [reply doc/man3/ASN1_INTEGER_new.pod:16]: Fixing this PR will be better, I think.

File: doc/man3/ASN1_INTEGER_new.pod (hunk 3 above)
Comment (NMorozxov) [reply doc/man3/ASN1_INTEGER_new.pod:5]: fixed

**Comment by NMorozxov** [comment]
fix commited

**Comment by NMorozxov** [comment]
can you restart a failed job?

**Comment by beldmit** [comment]
Done

File: doc/man3/ASN1_INTEGER_new.pod (hunk 5 above)
Comment (mattcaswell) [reply doc/man3/ASN1_INTEGER_new.pod:16]: > i must drop this page and combit it with others ? Or I must fix this PR

Fixing this PR to add a new "generic" page for all ASN1 *_new/*_free functions would be the ideal solution.

Note - I'm not absolutely requiring you to do this as part of this PR if you don't want to. It's a suggestion for how we could make this much better - but its a fair amount of work to do - and what you have here is better than what we have at the moment (i.e. nothing). So if you don't want to make a generic page - then fine. I will review this PR for what it is and maybe someone else can make that change later.

File: doc/man3/ASN1_INTEGER_new.pod (hunk 3 above)
Comment (mattcaswell) [reply doc/man3/ASN1_INTEGER_new.pod:5]: I don't see a fix for this?

Comment (NMorozxov) [reply doc/man3/ASN1_INTEGER_new.pod]: fix commited

**Comment by NMorozxov** [comment]
can you restart a failed job?

File: doc/man3/ASN1_INTEGER_new.pod (hunk 4 above)
Comment (NMorozxov) [reply doc/man3/ASN1_INTEGER_new.pod:18]: fixed

**Comment by beldmit** [comment]
Merged. Thanks!

**Pull request closed without merging**